lib/elixir/lib/supervisor.ex

```diff
@@ -543,7 +546,7 @@ defmodule Supervisor do
 
   If the supervisor and its child processes are successfully spawned
   (if the start function of each child process returns `{:ok, child}`,
-  `{:ok, child, info}`, or `:ignore`) this function returns
+  `{:ok, child, info}`, or `:ignore`) or if `children` is an empty list, this function returns
```

Comment:
```suggestion
  `{:ok, child, info}`, or `:ignore`), this function returns
```

Reply:
By saying that "all child processes are successfully spawned" does not cover the "no child process is spawned".
It would be interpreted as that would return `:error`

Reply:
All the quantifier typically covers none as the truth predicate. `Enum.all?([], & &1) == true`. This is typically how the documentation is written and I am not sure we should denote empty lists exceptions everywhere.

Reply:
We could also rewrite to say: none of its child fail to start by returning `{:error, reason}`. But both all and none are clear predicates.

Reply:
> "All", the quantifier, typically covers empty as the truth predicate. `Enum.all?([], & &1) == true`.

This may not be totally understood in layman terms. In any case going with the negated version of it (`{:error, reason}` while correct and clear it will make it more complicated I guess.